Source/ARTRealtime.m

```diff
@@ -268,11 +282,12 @@ - (void)transition:(ARTRealtimeConnectionState)state withErrorInfo:(ARTErrorInfo
 
 - (void)transitionSideEffects:(ARTConnectionStateChange *)stateChange {
     ARTStatus *status = nil;
+    __weak __typeof(self) weakSelf = self;
```

Comment:
Yes, you're right, I don't want to increase the ref count on those blocks because, for example, the `unlessStateChangesBefore` is setting a timer and if the `ARTRealtime` instance is released before that timer, then it could create a retain cycle and then we have a leak. That could also bring some instability in the test suite because we are creating a lot of `ARTRealtime` instances. I will add a comment about it. Thanks.